docs/examples/Parameters/Parameters.ipynb

```diff
@@ -141,6 +134,159 @@
+   "outputs": [],
+   "source": [
+    "class VirtualParameter(Parameter):\n",
+    "    def __init__(self, name, dac_param):\n",
+    "        self._dac_param = dac_param\n",
+    "        super().__init__(name)\n",
+    "        \n",
+    "    def get_raw(self):\n",
+    "        return self._dac_param.get()    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-06-17T08:58:48.861043Z",
+     "start_time": "2021-06-17T08:58:48.853047Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "vp1 = VirtualParameter('param_without_underlying_instrument_knowledge', dac.ch1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-06-17T08:58:49.967319Z",
+     "start_time": "2021-06-17T08:58:49.954319Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "vp1.underlying_instrument"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Right now, parameter `vp1` has no knowledge of the instrument of the parameter it wraps around. This can be addressed as follows."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-06-17T08:58:52.491912Z",
+     "start_time": "2021-06-17T08:58:52.473919Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "class NewVirtualParameter(Parameter):\n",
+    "    def __init__(self, name, dac_param):\n",
+    "        self._dac_param = dac_param\n",
+    "        super().__init__(name)\n",
+    "    \n",
+    "    @property\n",
+    "    def underlying_instrument(self) -> Optional[InstrumentBase]:\n",
+    "        return self._dac_param.root_instrument\n",
+    "    \n",
+    "    def get_raw(self):\n",
+    "        return self._dac_param.get()    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-06-17T08:59:00.598989Z",
+     "start_time": "2021-06-17T08:59:00.583959Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "vp2 = NewVirtualParameter('param_with_underlying_instrument_knowledge', dac.ch1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-06-17T08:59:06.965786Z",
+     "start_time": "2021-06-17T08:59:06.943669Z"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<DummyInstrument: dac>"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "vp2.underlying_instrument"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, `vp2` parameter knows that its underlying instrument is `dac`. This is quite useful while measuring these virtual parameters with `doNd` functions with `use_threads` flag set to `True`.   "
```

Comment:
Should we explain here why that is ?

Reply:
Explain why param knows about underlying instrument or why is it useful?

Reply:
Got it, explaining why it is useful.